sunpy/map/tests/test_mapbase.py

```diff
@@ -613,3 +613,20 @@ def test_draw_contours_noerror(aia171_test_map):
 def test_draw_contours_aia(aia171_test_map):
     aia171_test_map.plot()
     aia171_test_map.draw_contours(u.Quantity(np.arange(1, 100, 10), 'percent'))
+
+def test_more_than_two_dimensions():
+    """Check to see if an appropriate error is provided when a FIT data with more than two dimensions is loaded
+    Fixes #1919
+    To replicate the bug we need to load a >2-dim dataset with a OBSERVATORY header
+    """
+    bad_data = np.random.rand(2,2,2)
+    hdr = fits.Header()
+    hdr['TELESCOP'] = 'XXX'
+    # Test fails if there is no user warning
+    warnings.simplefilter('always')
+    pytest.warns(Warning,'sunpy.map.Map(bad_data, hdr)')
+    # Test fails if the bad_map.__repr__() causes an exception
+    bad_map = sunpy.map.Map(bad_data, hdr)
+    repr(bad_map)
```

Comment:
I think the best way might be to test that `bad_map.dimensions` gives the correct numerical output using `astropy.tests.helper.assert_quantity_all_close`